src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -418,13 +419,30 @@ public void Flush()
             this.configuration.TelemetryChannel.Flush();
         }
 
-        private void WriteTelemetryToDebugOutput(ITelemetry telemetry)
+        private void WriteTelemetryToDebugOutput(ITelemetry telemetry, bool isConfigured)
         {
             if (this.debugOutput.IsAttached() && this.debugOutput.IsLogging())
             {
-                string prefix = string.IsNullOrEmpty(telemetry.Context.InstrumentationKey) ?
-                    "Application Insights Telemetry (unconfigured): " :
-                    "Application Insights Telemetry: ";
+                InternalTelemetryProperties properties = null;
+                Utils.ExtraTelemetryProperties.TryGetValue(telemetry, out properties);
```

Comment:
Let's remove it after we read it once. Just in case customer wants to reuse the same item later since we do not force them not to do it, I know it is not thread safe, but it's better than nothing =)